controllers/users.js

```diff
@@ -362,18 +362,25 @@ const getSelfDetails = async (req, res) => {
 const updateSelf = async (req, res) => {
   try {
     const { id: userId } = req.userData;
+    const { user } = await dataAccess.retrieveUsers({ id: userId });
+
     if (req.body.username) {
-      const { user } = await dataAccess.retrieveUsers({ id: userId });
       if (!user.incompleteUserDetails) {
         return res.boom.forbidden("Cannot update username again");
       }
       await userQuery.setIncompleteUserDetails(userId);
     }
 
-    const user = await userQuery.addOrUpdate(req.body, userId);
+    if (req.body.roles) {
+      if (user && user.roles.in_discord) {
```

Comment:
should we not add developers role as well here?

Reply:
why? please explain, so let's suppose I get kicked out of the group, I know that maven, product manager and developers tag allows me to directly generate link, I hit that API with those roles, and get a link for it, and then I change it

Reply:
Added
thanks for pointing this out